library/cwm/src/lib/cwm/multi_status_selector.rb

```diff
@@ -0,0 +1,470 @@
+      end
+
+      # Whether item is auto selected
+      #
+      # @return [Boolean] true if the status is auto selected; false otherwise
+      def auto_selected?
+        status == AUTO_SELECTED
+      end
+
+      # Sets the item as selected
+      def select!
+        @status = SELECTED
+      end
+
+      # Sets the item as not selected
+      def unselect!
+        @status = UNSELECTED
+      end
+
+      # Sets the item as auto-selected
+      def auto_select!
+        @status = AUTO_SELECTED
+      end
+
+      # Returns richtext representation for the item
+      #
+      # Basically, an string containing two <a> or <span> tags, depending on the #enabled? method.
+      # One for the check box input and another for the label.
+      #
+      # @return [String] the item richtext representation
+      def to_s
+        "#{checkbox_input} #{checkbox_label}"
+      end
+
+    private
+
+      # @see .icon_for
+      def icon
+        self.class.icon_for(status, mode: mode, state: state)
+      end
+
+      # Builds the check box input representation
+      #
+      # @return [String]
+      def checkbox_input
+        if enabled?
+          "<a href=\"#{id}#{INPUT_EVENT_ID}\" style=\"#{text_style}\">#{icon}</a>"
+        else
+          "<span style\"#{text_style}\">#{icon}</a>"
+        end
+      end
+
+      # Builds the check box label representation
+      #
+      # @return [String]
+      def checkbox_label
+        if enabled?
+          "<a href=\"#{id}#{LABEL_EVENT_ID}\" style=\"#{text_style}\">#{label}</a>"
+        else
+          "<span style\"#{text_style}\">#{label}</a>"
+        end
+      end
+
+      # Returns the current mode
+      #
+      # @return [String] "normal" in a running system; "inst" during the installation
+      def mode
+        installation? ? "inst" : "normal"
+      end
+
+      # Returns the current input state
+      #
+      # @return [String] "enabled" when item must be enabled; "disabled" otherwise
+      def state
+        enabled? ? "enabled" : "disabled"
+      end
+
+      # Returns style rules for the text
+      #
+      # @return [String] the status text representation
+      def text_style
+        "text-decoration: none; color: #{color}"
+      end
+
+      # Determines the color for the text
+      #
+      # @return [String] "grey" for a disabled item;
+      #                  "white" when enabled and running in installation mode;
+      #                  "black" otherwise
+      def color
+        return "grey" unless enabled?
+        return "white" if installation?
+
+        "black"
+      end
+
+      # Determines whether running in installation mode
+      #
+      # We do not use Stage.initial because of firstboot, which runs in 'installation' mode
+      # but in 'firstboot' stage.
+      #
+      # @return [Boolean] Boolean if running in installation or update mode
+      def installation?
+        Yast::Mode.installation || Yast::Mode.update
+      end
+    end
```

Comment:
Question: could we simply use a `class` attribute for those links and properly style them via .qss? Or is not possible because they are inside a RichText?

@lslezak, @shundhammer :point_up_2: :point_up_2: